### PR TITLE
Updated withdrawl logic with random shuffle

### PR DESCRIPTION
**Pull request by vedantvasugupta**

### CODE REVIEW

**Comment by kishan-v** [comment]
@vedantvasugupta Please base the withdrawal order shuffling on the reworked RunTurn in #31

File: server/EnvironmentServer.go [hunk 1]
```diff
@@ -27,15 +27,55 @@ type EnvironmentServer struct {
...
+						withdrawal, agentID, teamPool)
+				}
+			}
+		}
+		// Update the team's pool value after processing all withdrawals
+		team.CommonPool = teamPool
+		cs.teams[team.TeamID] = team
+	}
+}
+
```
Comment (kishan-v) [inline server/EnvironmentServer.go:67]: Please see my [comment](https://github.com/ADimoska/SOMASExtended/pull/32#issuecomment-2500814082)

**Review by kishan-v — CHANGES_REQUESTED** [review]

File: server/EnvironmentServer.go (hunk 1 above)
Comment (vedantvasugupta) [reply server/EnvironmentServer.go:67]: Hi @kishan-v  that branch has been deleted

**Comment by vedantvasugupta** [comment]
Opening a new pr for this

**Pull request closed without merging**